src/commands/serve.rs

```diff
@@ -357,3 +391,55 @@ impl hyper::service::Service<Request> for ProxyHandler {
         })
     }
 }
+
+#[derive(Clone)]
+struct LogStream {
+    prefix: String,
+    output: OutputMutex,
+}
+
+impl preview2::StdoutStream for LogStream {
+    fn stream(&self) -> Box<dyn preview2::HostOutputStream> {
+        Box::new(self.clone())
+    }
+
+    fn isatty(&self) -> bool {
+        false
+    }
+}
+
+impl preview2::HostOutputStream for LogStream {
+    fn write(&mut self, bytes: bytes::Bytes) -> StreamResult<()> {
+        let mut msg = Vec::new();
+
+        for line in bytes.split(|c| *c == b'\n') {
+            if !line.is_empty() {
+                msg.extend_from_slice(&self.prefix.as_bytes());
+                msg.extend_from_slice(line);
+                msg.push(b'\n');
+            }
+        }
+
+        let output = self.output.clone();
+        tokio::task::spawn(async move {
+            use std::io::Write;
+            let mut output = output.lock().await;
+            output.write_all(&msg).expect("writing to stdout");
+        });
+
+        Ok(())
+    }
+
+    fn flush(&mut self) -> StreamResult<()> {
+        Ok(())
+    }
+
+    fn check_write(&mut self) -> StreamResult<usize> {
+        Ok(1024 * 1024)
+    }
+}
+
+#[async_trait::async_trait]
+impl preview2::Subscribe for LogStream {
+    async fn ready(&mut self) {}
```

Comment:
It's not clear that there's a reasonable implementation for this here. Perhaps using `tokio::io::Stdout` instead of a mutex guarded `std::io::Stdout` would make more sense?